addons/openacademy/models/session.py

```
@@ -1,14 +1,71 @@
+    instructor_id = fields.Many2one('res.partner', ondelete='set null')
+    attendees = fields.Many2many('res.partner', 'session_attendees', string="Attendees")
+
+    description = fields.Html(string='Description', related="course_id.description")
+
+    seats_taken = fields.Integer(string="Seats taken", compute='_compute_seats_taken')
+    is_full = fields.Boolean(string="Course is Full", compute='_compute_is_full', store=True)
+
+    @api.depends('start_date')
+    def _compute_end_date(self):
+        for record in self:
+            record.end_date = record.start_date + timedelta(minutes=record.duration)
+
+    @api.depends('attendees')
+    def _compute_seats_taken(self):
+        for record in self:
+            if record.attendees:
+                record.seats_taken = len(record.attendees) / record.seats * 100
+            else:
+                record.seats_taken = 0
+
+    @api.depends('attendees')
+    def _compute_is_full(self):
+        for record in self:
+            record.is_full = len(record.attendees) == record.seats
+
+    @api.onchange("seats")
+    def _check_seats(self):  # self is not a collection
+        if self.seats < 0:
+            return {
+                'warning': {
+                    'title': "Negative Seats",
+                    'message': "You can't have negativ Seats",
+                }
+            }
+
+    @api.onchange('attendees', 'seats')
+    def _check_available_seats(self):  # self is not a collection
+        if self.seats_taken > 100:
+            return {
+                'warning': {
+                    'title': "Session Full",
+                    'message': "The Session is full",
+                }
+            }
+
+    @api.constrains('instructor_id', 'attendees')
+    def _check_instructor_is_not_an_attendee(self):
+        for record in self:
+            for attendee in record.attendees:
+                if attendee.id == record.instructor_id.id:
+                    return {
```

Here this is not the way to raise a Warning. 
You should instead raise an Error, with any of the classes Odoo defines for Errors like ValidationError, UserError, etc. 
Look for them in the code to find some examples